conan/tools/meson/toolchain.py

```diff
@@ -117,6 +123,8 @@ def __init__(self, conanfile, backend=None):
             if is_apple_os(os_host):  # default cross-compiler in Apple is common
                 default_comp = "clang"
                 default_comp_cpp = "clang++"
+            if (arch_host != arch_build) or (os_host == "iOS") or (os_host == "Android" and os_build != "Android"):
```

Comment:
There is a helper that is called ``cross_building()``. Maybe it is better to use it here? This seems a bit ad-hoc, assuming that iOS and Android is cross building. Can you please clarify a bit the logic when this property is needed?

Reply:
`needs_exe_wrapper` is required whenever the compiled binary for the host is incapable of running on the build machine. Meson conducts compiler sanity checks and attempts to determine this itself. However, it will fail in certain cases, like armv8 iOS trying to run on Macos armv8. Meson thinks it makes sense to try and run the armv8 iOS binary because the build machine is also armv8, but it fails. This option is needed to suppress that check.

This method will add needs_exe_wrapper in other cases where meson can successfully autodetect, as well. I don't think that hurts anything, but perhaps it is overly explicit. For example, meson can tell that it shouldn't try to run an iOS x86_64 binary even if the option needs_exe_wrapper is not set. So it would be redundant (but still present) in that case. 

I don't have a problem simply using `cross_building()`. I'll go take a look at the integration tests and try to write some meaningful tests tonight.

Reply:
It does look like that branch of code is executed is things are cross_building anyways (`if cross_building(conanfile, skip_x86_64=True`), so I could simply always include `needs_exe_wrapper` if that block of code executes. But I'm not quite sure what `skip_x64_x86` does. Is that simply saying that compiling from 64 bit to 32 bit is not considered crosscompiling?

Reply:
> But I'm not quite sure what skip_x64_x86 does. Is that simply saying that compiling from 64 bit to 32 bit is not considered crosscompiling?

Yes, exactly, that means that a 32 bits ``x86`` binary in Windows runs fine in Windows x86_64. It is generally not considered cross-building in Windows builds.